base/applications/utilman/dialog.c

```diff
@@ -0,0 +1,454 @@
+                               0,
+                               0,
+                               LR_DEFAULTSIZE);
+
+    /* Set the icon within the dialog's title bar */
+    if (Globals.hIcon)
+    {
+        SendMessageW(hDlg, WM_SETICON, ICON_SMALL, (LPARAM)Globals.hIcon);
+    }
+
+    /* Retrieve the system menu and append the "About" menu item onto it */
+    hSysMenu = GetSystemMenu(hDlg, FALSE);
+    if (hSysMenu != NULL)
+    {
+        if (LoadStringW(Globals.hInstance, IDM_ABOUT, szAboutDlg, _countof(szAboutDlg)))
+        {
+            AppendMenuW(hSysMenu, MF_SEPARATOR, 0, NULL);
+            AppendMenuW(hSysMenu, MF_STRING, IDM_ABOUT, szAboutDlg);
+        }
+    }
+
+    /* Get the dialog items, specifically the dialog list box, the Start and Stop buttons */
+    Globals.hListDlg = GetDlgItem(hDlg, IDC_LISTBOX);
+    Globals.hDlgCtlStart = GetDlgItem(hDlg, IDC_START);
+    Globals.hDlgCtlStop = GetDlgItem(hDlg, IDC_STOP);
+
+    /* Set the selection to the first item (Magnify) */
+    Globals.iSelectedIndex = 0;
+
+    /* Refresh the list */
+    ListBoxRefreshContents();
+
+    /* Load the groupbox's title string */
+    LoadStringW(Globals.hInstance, IDC_GROUPBOX_OPTIONS_TITLE, EntriesList[0].szCaptionUtility, _countof(EntriesList[0].szCaptionUtility));
+
+    /*
+        Check that the characters are actually copied into the buffer, then append
+        the string using StringCchPrintfW() to the format.
+    */
+    if (LoadStringW(Globals.hInstance, IDC_MAGNIFIER, Globals.szResource, _countof(Globals.szResource)))
+    {
+        StringCchPrintfW(Globals.szFormat, _countof(Globals.szFormat), EntriesList[0].szCaptionUtility, Globals.szResource);
+    }
+
+    /* Add the formatted string to the groupbox's title */
+    SetWindowTextW(GetDlgItem(hDlg, IDC_GROUPBOX), Globals.szFormat);
+    
+    /* Create a timer, we'll use it to control the state of our items in the listbox */
+    Globals.iTimer = SetTimer(hDlg, 0, 1000, NULL);
+
+    return TRUE;
+}
+
+/**
+ * @CheckUtilityState
+ * @implemented
+ *
+ * Checks the state of the given accessibility tool.
+ *
+ * @param bUtilState
+ * State condition (in boolean of type INT).
+ *
+ * @return
+ * Nothing.
+ *
+ */
+VOID CheckUtilityState(BOOL bUtilState)
+{
+    if (bUtilState == TRUE)
+    {
+        Button_Enable(Globals.hDlgCtlStart, FALSE);
+        Button_Enable(Globals.hDlgCtlStop, TRUE);
+    }
+    else
+    {
+        Button_Enable(Globals.hDlgCtlStart, TRUE);
+        Button_Enable(Globals.hDlgCtlStop, FALSE);
+    }
+}
+
+/**
+ * @ListBoxRefreshContents
+ * @implemented
+ *
+ * Handle the tasks on a periodic cycle. This function handles WM_TIMER message.
+ *
+ * @return
+ * Returns 0 to inform the system that WM_TIMER has been processed.
+ *
+ */
+INT ListBoxRefreshContents(VOID)
+{
+    /* Remove all the items from the listbox */
+    SendMessageW(Globals.hListDlg, LB_RESETCONTENT, 0, 0);
+
+    if (!IsProcessRunning(EntriesList[0].lpProgram))
```

Comment:
What is the point of an array if you are going to use hardcoded indexes here?

Reply:
The next improvement to be done will be to **not** use the hardcoded indices, but instead loop over the list of available accessibility utilities handled by utilman in order to check their real-time status.